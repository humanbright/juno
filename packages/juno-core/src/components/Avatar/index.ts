export * from './Avatar';
export * from './utils/useAvatarColorToken';
export * from './utils/useAvatarShortName';
