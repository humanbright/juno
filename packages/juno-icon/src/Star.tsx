import * as React from 'react';
import { SVGProps, Ref, forwardRef, memo } from 'react';

const Star = memo(
  forwardRef((props: SVGProps<SVGSVGElement>, ref: Ref<SVGSVGElement>) => (
    <svg
      viewBox="0 0 32 32"
      xmlns="http://www.w3.org/2000/svg"
      ref={ref}
      {...props}
    >
      <path d="M16.885 1.726c.394.195.714.514.908.908l2.97 6.018a1 1 0 0 0 .753.547l6.641.965a2.001 2.001 0 0 1 1.109 3.412L24.46 18.26c-.236.23-.343.561-.288.885l1.134 6.614a1.999 1.999 0 0 1-2.902 2.108l-5.94-3.123a1.002 1.002 0 0 0-.931 0l-5.94 3.123a2 2 0 0 1-2.902-2.109l1.134-6.614a.999.999 0 0 0-.288-.885l-4.806-4.684a2 2 0 0 1 1.109-3.411l6.641-.965a1 1 0 0 0 .753-.547l2.97-6.018a2 2 0 0 1 2.679-.908z" />
    </svg>
  )),
);
Star.displayName = 'Star';
Star['iconName'] = 'star';
export default Star;
