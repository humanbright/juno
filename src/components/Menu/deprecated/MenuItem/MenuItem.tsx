import { MenuItemProps as MuiMenuItemProps } from '@material-ui/core/MenuItem';
import clsx from 'clsx';
import uniqueId from 'lodash/uniqueId';
import React, {
  forwardRef,
  ReactNode,
  useContext,
  useEffect,
  useMemo,
  useState,
} from 'react';

import {
  Classes,
  combineClasses,
  omit,
  RcBaseProps,
  RcBaseSize,
  RcClassesProps,
  styled,
  useEventCallback,
  withDeprecatedCheck,
} from '../../../../foundation';
import checkSvg from '../../../../icon/Check';
import { RcIcon, RcIconProps } from '../../../Icon';
import { withTooltip } from '../../../Tooltip';
import { RcMenuContext } from '../Menu';
import { RcMenuListContext } from '../MenuList';
import { RcSubMenuContext } from '../SubMenu';
import {
  StyledMenuItem,
  StyledMuiListItemIcon,
  StyledRcIcon,
  StyledRcListItemLabel,
} from './styles';
import { RcMenuItemClasses } from './utils/MenuItemUtil';

type MenuItemSize = RcBaseSize<'large' | 'medium'>;

type RcMenuItemType = 'checked' | 'selected';

type RcMenuItemClassesType = RcClassesProps<
  Classes<MuiMenuItemProps> | 'checked' | 'unchecked'
>;

type RcMenuItemProps = {
  /** MenuItem size */
  size?: MenuItemSize;
  /** MenuItem with subtext */
  subtext?: ReactNode;

  /** checked for that should render checked icon */
  type?: RcMenuItemType;
  /** if type to be `checked`, the checked prop for that should render checked icon */
  checked?: boolean;

  /** @deprecated  MenuItem with icon, pls use RcListItemIcon */
  icon?: ReactNode;
  /** @deprecated pass to the icon, pls use RcIcon with symbol prop */
  symbol?: RcIconProps['symbol'];
  /** @deprecated MenuItem with avatar, pls use ListItemAvatar */
  avatar?: React.ReactElement;
  /** @deprecated */
  label?: string;

  /** @deprecated the maxWidth of MenuItem, recommend pass classes to define it. */
  maxWidth?: number;
  /**
   * @deprecated
   * when hover on MenuItem, is cursor in shape of pointer.
   * Instead recommend using classes to define it.
   */
  isMember?: boolean;
  /**  @deprecated automationId, can use 'data-test-automation-id' replace */
  automationId?: string;
} & RcMenuItemClassesType &
  RcBaseProps<
    MuiMenuItemProps,
    'disableGutters' | 'button' | 'classes' | 'title' | 'dense' | 'divider'
  >;

const _RcMenuItem = forwardRef<any, RcMenuItemProps>((props, ref) => {
  const {
    icon,
    subtext,
    type,
    checked,
    value,
    symbol,
    children,
    disabled,
    classes,
    isMember,
    avatar,
    automationId,
    maxWidth,
    size,
    label,
    className,
    onMouseEnter,
    onClick,
    ...rest
  } = props;

  const isCheckedType = type === 'checked';

  const iconElement = useMemo(() => {
    if (React.isValidElement(avatar)) {
      return avatar;
    }
    if (symbol) {
      return <RcIcon size="small" symbol={symbol} />;
    }
    if (React.isValidElement(icon)) {
      return icon;
    }
    if (typeof icon === 'string') {
      return <RcIcon size="small">{icon}</RcIcon>;
    }
    return null;
  }, [avatar, icon, symbol]);

  const renderSubText = useMemo(() => {
    if (!isCheckedType) {
      return subtext;
    }
    return checked ? (
      <StyledRcIcon
        size="large"
        symbol={checkSvg}
        color={['interactive', 'f01']}
      />
    ) : (
      subtext
    );
  }, [checked, isCheckedType, subtext]);

  const [menuItemUUID, setMenuItemUUID] = useState<string | undefined>(
    undefined,
  );

  const menuContext = useContext(RcMenuContext);
  const menuListContext = useContext(RcMenuListContext);
  const subMenuContext = useContext(RcSubMenuContext);

  useEffect(() => {
    if (!menuListContext.isMenuList) {
      menuListContext.setIsMenuList();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
    setMenuItemUUID(uniqueId('rc-menu-item-'));
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const handleMenuItemMouseEnter = useEventCallback(
    (e: React.MouseEvent<HTMLLIElement>) => {
      onMouseEnter && onMouseEnter(e);

      if (!menuItemUUID) {
        return;
      }
      if (menuListContext.isMenuList) {
        menuListContext.setFocusedMenuItemId(menuItemUUID);
      } else {
        menuContext.setFocusedMenuItemId(menuItemUUID);
      }
    },
  );

  const handleMenuItemClick = useEventCallback(
    (e: React.MouseEvent<HTMLLIElement>) => {
      onClick && onClick(e);
      if (menuContext.autoClose) {
        menuContext.closeMenu(e, 'tabKeyDown');
      }
      if (subMenuContext.autoClose && subMenuContext.closeSubMenu) {
        subMenuContext.closeSubMenu();
      }
    },
  );

  const groupClasses = useMemo(
    () => combineClasses(RcMenuItemClasses, classes),
    [classes],
  );

  const toClasses = useMemo(
    () => omit(groupClasses!, ['checked', 'unchecked']),
    [groupClasses],
  );

  return (
    <StyledMenuItem
      data-test-automation-id={automationId}
      disabled={disabled}
      isMember={isMember}
      data-disabled={disabled}
      maxWidth={maxWidth}
      ref={ref}
      size={size}
      value={value}
      {...rest}
      classes={toClasses}
      onMouseEnter={handleMenuItemMouseEnter}
      onClick={handleMenuItemClick}
      className={clsx(className, {
        [groupClasses!.checked]: isCheckedType && checked,
        [groupClasses!.unchecked]: isCheckedType && !checked,
      })}
    >
      {iconElement && (
        <StyledMuiListItemIcon>{iconElement}</StyledMuiListItemIcon>
      )}
      {label && <StyledRcListItemLabel>{label}&nbsp;</StyledRcListItemLabel>}
      {children}
      {renderSubText}
    </StyledMenuItem>
  );
});

/** @release */
const RcMenuItem = styled(
  withDeprecatedCheck(
    withTooltip(_RcMenuItem),
    [
      {
        prop: 'isMember',
        time: '2021-1',
        comment:
          'isMember used for when hover on MenuItem, is cursor in shape of pointer. Please instead recommend using classes to define it.',
      },
      {
        prop: 'maxWidth',
        time: '2021-1',
        comment: 'Please instead recommend using classes to define it.',
      },
      {
        prop: 'label',
        time: '2021-1',
        comment: 'Please add label component and add to children prop',
      },
    ],
    'RcMenuItem',
  ),
)``;

RcMenuItem.displayName = 'RcMenuItem';

RcMenuItem.defaultProps = {
  size: 'medium',
};

export { RcMenuItem, RcMenuItemProps, RcMenuItemType, RcMenuItemClassesType };
