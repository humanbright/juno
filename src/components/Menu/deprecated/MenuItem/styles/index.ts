export * from './StyledMenuItem';
export * from './StyledMuiListItemIcon';
export * from './StyledRcListItemLabel';
