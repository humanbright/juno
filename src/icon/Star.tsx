import React, { forwardRef, memo } from 'react';

const Star = memo(
  forwardRef(
    (
      props: React.SVGProps<SVGSVGElement>,
      svgRef?: React.Ref<SVGSVGElement>,
    ) => (
      <svg
        viewBox="0 0 32 32"
        xmlns="http://www.w3.org/2000/svg"
        ref={svgRef}
        {...props}
      >
        <path d="M16.885 1.726c.394.195.714.514.908.908l2.97 6.018a1 1 0 00.753.547l6.641.965a2.001 2.001 0 011.109 3.412L24.46 18.26c-.236.23-.343.561-.288.885l1.134 6.614a1.999 1.999 0 01-2.902 2.108l-5.94-3.123a1.002 1.002 0 00-.931 0l-5.94 3.123a2 2 0 01-2.902-2.109l1.134-6.614a.999.999 0 00-.288-.885l-4.806-4.684a2 2 0 011.109-3.411l6.641-.965a1 1 0 00.753-.547l2.97-6.018a2 2 0 012.679-.908z" />
      </svg>
    ),
  ),
);
Star.displayName = 'Star';
Star['iconName'] = 'star';
export default Star;
