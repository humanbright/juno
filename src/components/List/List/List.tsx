import MuiList from '@material-ui/core/List';
import React, { ComponentProps, forwardRef, useMemo } from 'react';

import {
  combineClasses,
  RcBaseProps,
  styled,
  useThemeProps,
} from '../../../foundation';
import { ListStyle } from './styles';
import { RcListClasses } from './utils';

type RcListProps = {} & RcBaseProps<ComponentProps<typeof MuiList>>;

const _RcList = forwardRef<any, RcListProps>((inProps, ref) => {
  const props = useThemeProps({ props: inProps, name: 'RcList' });
  const { classes: classesProp, children, ...rest } = props;
  const classes = useMemo(
    () => combineClasses(RcListClasses, classesProp),
    [classesProp],
  );

  return (
    <MuiList {...rest} ref={ref} classes={classes}>
      {children}
    </MuiList>
  );
});

const RcList = styled(_RcList)`
  ${ListStyle}
`;

RcList.defaultProps = {
  disablePadding: true,
};

RcList.displayName = 'RcList';

export { RcList };
export type { RcListProps };
