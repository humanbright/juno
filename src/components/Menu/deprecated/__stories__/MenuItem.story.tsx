import { boolean, text } from '@storybook/addon-knobs';
import { Meta, Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';

import { RcMenuItem } from '../MenuItem';
import Star from '../../../../icon/Star';
import {
  notControlInDocTable,
  notShowInDocTable,
  sortInDocTable,
} from '../../../../storybook';
import { RcAvatar } from '../../../Avatar';
import avatar from '../../../Avatar/__stories__/img/avatar.jpg';
import { RcListItemText } from '../../../List';

export default {
  title: 'ðŸ–¤ deprecated Components/Menus/MenuItem',
  component: RcMenuItem,
  argTypes: {
    ...sortInDocTable<keyof MenuItemProps>([
      'size',
      'type',
      'disabled',
      'selected',
      'checked',
      'autoFocus',
      'useRcTooltip',
      'title',
      'alignItems',
    ]),
    ...notShowInDocTable<keyof MenuItemProps>([
      'style',
      'innerRef',
      'automationId',
      'maxWidth',
      'isMember',
      'label',
    ]),
    ...notControlInDocTable<keyof MenuItemProps>([
      'icon',
      'avatar',
      'symbol',
      'subtext',
    ]),
  },
} as Meta;

const Avatar = <RcAvatar src={avatar} />;

type MenuItemProps = ComponentProps<typeof RcMenuItem>;

export const MenuItem: Story<MenuItemProps> = ({ ...args }) => {
  return (
    <>
      <RcMenuItem onClick={() => console.log('onClick Profile')} {...args}>
        Profile
      </RcMenuItem>
      <RcMenuItem onClick={() => console.log('onClick Profile')} {...args}>
        Profile
      </RcMenuItem>
    </>
  );
};

MenuItem.storyName = 'MenuItem';

export const TextMenuItemWithIcon = () => {
  return (
    <RcMenuItem onClick={() => console.log('onClick Profile')} symbol={Star}>
      Profile
    </RcMenuItem>
  );
};

TextMenuItemWithIcon.story = {
  name: 'Text MenuItem With Icon',
};

export const TextMenuItemWithLabel = () => {
  return <RcMenuItem label={'label1:'}>test@ringcentral.com</RcMenuItem>;
};

TextMenuItemWithLabel.story = {
  name: 'Text MenuItem With Label',
};

export const _2LinesMenuItemWithAvatar = () => {
  const primary = text('primary', 'Two-line item name');
  const secondaryText = text('secondaryText', 'Secondary text');
  return (
    <RcMenuItem onClick={() => console.log('onClick Profile')} avatar={Avatar}>
      <RcListItemText primary={primary} secondary={secondaryText} />
    </RcMenuItem>
  );
};

_2LinesMenuItemWithAvatar.story = {
  name: '2 Lines MenuItem With Avatar',
};

export const MenuItemWithSubtext = () => {
  const primary = text('primary', 'Profile');
  const subtext = text('subtext', 'Subtext');

  return (
    <RcMenuItem
      onClick={() => console.log('onClick Profile')}
      subtext={subtext}
    >
      <RcListItemText primary={primary} />
    </RcMenuItem>
  );
};

MenuItemWithSubtext.story = {
  name: 'MenuItem with subtext',
};

export const CheckedMenuItem = () => {
  const checked = boolean('checked', false);
  const primary = text('primary', 'Profile');

  return (
    <>
      <br />
      <br />
      <RcMenuItem
        onClick={() => console.log('onClick Profile')}
        value={'test'}
        type="checked"
        checked={checked}
        TooltipProps={{}}
      >
        <RcListItemText primary={primary} />
      </RcMenuItem>
    </>
  );
};

CheckedMenuItem.story = {
  name: 'Checked MenuItem',
};
