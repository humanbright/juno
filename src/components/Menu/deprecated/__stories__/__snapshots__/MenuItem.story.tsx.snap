// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ðŸ–¤ deprecated Components/Menus/MenuItem 2 Lines MenuItem With Avatar 1`] = `
.c4 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  outline: none;
  color: #FFFFFF;
  font-size: 1.25rem;
  font-weight: 700;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 28px;
  width: 64px;
  height: 64px;
  background-color: #F3F3F3;
}

.c4 > img,
.c4 > canvas {
  width: 100%;
  height: 100%;
}

.c3 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  outline: 0;
}

.c5 {
  padding: 0;
  margin: 0 8px 0 0;
}

.c5.RcListItemText-multiline {
  margin: 4px 8px 4px 0px;
}

.c5.RcListItemText-inset {
  padding-left: 56px;
}

.c5 .RcListItemText-primary {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-text-decoration-line: inherit;
  text-decoration-line: inherit;
  font-size: 0.9375rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 22px;
  color: #121212;
}

.c5 .RcListItemText-secondary {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-text-decoration-line: inherit;
  text-decoration-line: inherit;
  font-size: 0.75rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 16px;
  color: #666666;
}

.c2 {
  min-width: unset;
  margin-right: 8px;
  font-size: 1rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 24px;
  color: #666666;
}

.c0 {
  outline: none;
  padding: 4px 16px;
  font-size: 0.9375rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 22px;
  color: #121212;
  height: auto;
  min-height: 32px;
  min-width: 112px;
  box-sizing: border-box;
}

.c0.RcMenuItem-unchecked {
  padding-right: 48px;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover .sc-cSiAOC {
  display: block;
}

.c0:active {
  background-color: #066FAC;
  color: #fff;
}

.c0:active .c1 {
  color: #fff;
}

<li
  aria-disabled={false}
  className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root c0 MuiListItem-button"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="menuitem"
  tabIndex={-1}
>
  <div
    className="MuiListItemIcon-root c1 c2"
  >
    <div
      className="c3 "
      data-test-automation-class="avatar"
    >
      <div
        className="c4 RcAvatar-avatarContainer"
      >
        <img
          src="test-file-stub"
        />
      </div>
    </div>
  </div>
  <div
    className="MuiListItemText-root c5 MuiListItemText-multiline RcListItemText-multiline"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary RcListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
    >
      Two-line item name
    </span>
    <p
      className="MuiTypography-root MuiListItemText-secondary RcListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
    >
      Secondary text
    </p>
  </div>
</li>
`;

exports[`Storyshots ðŸ–¤ deprecated Components/Menus/MenuItem Checked MenuItem 1`] = `
Array [
  <br />,
  <br />,
  .c1 {
  padding: 0;
  margin: 0 8px 0 0;
}

.c1.RcListItemText-multiline {
  margin: 4px 8px 4px 0px;
}

.c1.RcListItemText-inset {
  padding-left: 56px;
}

.c1 .RcListItemText-primary {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-text-decoration-line: inherit;
  text-decoration-line: inherit;
  font-size: 0.9375rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 22px;
  color: #121212;
}

.c1 .RcListItemText-secondary {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-text-decoration-line: inherit;
  text-decoration-line: inherit;
  font-size: 0.75rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 16px;
  color: #666666;
}

.c0 {
  outline: none;
  padding: 4px 16px;
  font-size: 0.9375rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 22px;
  color: #121212;
  height: auto;
  min-height: 32px;
  min-width: 112px;
  box-sizing: border-box;
}

.c0.RcMenuItem-unchecked {
  padding-right: 48px;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover .sc-cSiAOC {
  display: block;
}

.c0:active {
  background-color: #066FAC;
  color: #fff;
}

.c0:active .sc-lgWdIC {
  color: #fff;
}

<li
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root c0 RcMenuItem-unchecked MuiListItem-button"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="menuitem"
    tabIndex={-1}
    value="test"
  >
    <div
      className="MuiListItemText-root c1"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary RcListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        Profile
      </span>
    </div>
  </li>,
]
`;

exports[`Storyshots ðŸ–¤ deprecated Components/Menus/MenuItem MenuItem 1`] = `
Array [
  .c0 {
  outline: none;
  padding: 4px 16px;
  font-size: 0.9375rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 22px;
  color: #121212;
  height: auto;
  min-height: 32px;
  min-width: 112px;
  box-sizing: border-box;
}

.c0.RcMenuItem-unchecked {
  padding-right: 48px;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover .sc-cSiAOC {
  display: block;
}

.c0:active {
  background-color: #066FAC;
  color: #fff;
}

.c0:active .sc-lgWdIC {
  color: #fff;
}

<li
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root c0 MuiListItem-button"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="menuitem"
    tabIndex={-1}
  >
    Profile
  </li>,
  .c0 {
  outline: none;
  padding: 4px 16px;
  font-size: 0.9375rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 22px;
  color: #121212;
  height: auto;
  min-height: 32px;
  min-width: 112px;
  box-sizing: border-box;
}

.c0.RcMenuItem-unchecked {
  padding-right: 48px;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover .sc-cSiAOC {
  display: block;
}

.c0:active {
  background-color: #066FAC;
  color: #fff;
}

.c0:active .sc-lgWdIC {
  color: #fff;
}

<li
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root c0 MuiListItem-button"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="menuitem"
    tabIndex={-1}
  >
    Profile
  </li>,
]
`;

exports[`Storyshots ðŸ–¤ deprecated Components/Menus/MenuItem MenuItem with subtext 1`] = `
.c1 {
  padding: 0;
  margin: 0 8px 0 0;
}

.c1.RcListItemText-multiline {
  margin: 4px 8px 4px 0px;
}

.c1.RcListItemText-inset {
  padding-left: 56px;
}

.c1 .RcListItemText-primary {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-text-decoration-line: inherit;
  text-decoration-line: inherit;
  font-size: 0.9375rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 22px;
  color: #121212;
}

.c1 .RcListItemText-secondary {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-text-decoration-line: inherit;
  text-decoration-line: inherit;
  font-size: 0.75rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 16px;
  color: #666666;
}

.c0 {
  outline: none;
  padding: 4px 16px;
  font-size: 0.9375rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 22px;
  color: #121212;
  height: auto;
  min-height: 32px;
  min-width: 112px;
  box-sizing: border-box;
}

.c0.RcMenuItem-unchecked {
  padding-right: 48px;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover .sc-cSiAOC {
  display: block;
}

.c0:active {
  background-color: #066FAC;
  color: #fff;
}

.c0:active .c2 {
  color: #fff;
}

<li
  aria-disabled={false}
  className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root c0 MuiListItem-button"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="menuitem"
  tabIndex={-1}
>
  <div
    className="MuiListItemText-root c1"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary RcListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
    >
      Profile
    </span>
  </div>
  Subtext
</li>
`;

exports[`Storyshots ðŸ–¤ deprecated Components/Menus/MenuItem Text MenuItem With Icon 1`] = `
.c4 {
  width: 1em;
  height: 1em;
  stroke-width: 0;
  stroke: currentColor;
  fill: currentColor;
  pointer-events: none;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 16px;
}

.c2 {
  min-width: unset;
  margin-right: 8px;
  font-size: 1rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 24px;
  color: #666666;
}

.c0 {
  outline: none;
  padding: 4px 16px;
  font-size: 0.9375rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 22px;
  color: #121212;
  height: auto;
  min-height: 32px;
  min-width: 112px;
  box-sizing: border-box;
}

.c0.RcMenuItem-unchecked {
  padding-right: 48px;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover .sc-cSiAOC {
  display: block;
}

.c0:active {
  background-color: #066FAC;
  color: #fff;
}

.c0:active .c1 {
  color: #fff;
}

<li
  aria-disabled={false}
  className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root c0 MuiListItem-button"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="menuitem"
  tabIndex={-1}
>
  <div
    className="MuiListItemIcon-root c1 c2"
  >
    <span
      className="c3 star icon"
    >
      <svg
        className="c4"
        viewBox="0 0 32 32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16.885 1.726c.394.195.714.514.908.908l2.97 6.018a1 1 0 00.753.547l6.641.965a2.001 2.001 0 011.109 3.412L24.46 18.26c-.236.23-.343.561-.288.885l1.134 6.614a1.999 1.999 0 01-2.902 2.108l-5.94-3.123a1.002 1.002 0 00-.931 0l-5.94 3.123a2 2 0 01-2.902-2.109l1.134-6.614a.999.999 0 00-.288-.885l-4.806-4.684a2 2 0 011.109-3.411l6.641-.965a1 1 0 00.753-.547l2.97-6.018a2 2 0 012.679-.908z"
        />
      </svg>
      
    </span>
  </div>
  Profile
</li>
`;

exports[`Storyshots ðŸ–¤ deprecated Components/Menus/MenuItem Text MenuItem With Label 1`] = `
.c0 {
  outline: none;
  padding: 4px 16px;
  font-size: 0.9375rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 22px;
  color: #121212;
  height: auto;
  min-height: 32px;
  min-width: 112px;
  box-sizing: border-box;
}

.c0.RcMenuItem-unchecked {
  padding-right: 48px;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover .sc-cSiAOC {
  display: block;
}

.c0:active {
  background-color: #066FAC;
  color: #fff;
}

.c0:active .c2 {
  color: #fff;
}

.c1 {
  font-weight: bold;
}

<li
  aria-disabled={false}
  className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root c0 MuiListItem-button"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="menuitem"
  tabIndex={-1}
>
  <span
    className="c1"
  >
    label1:
    Â 
  </span>
  test@ringcentral.com
</li>
`;
